### PR TITLE
Implement: Revamp - Bug report UI #1189

### PR DESCRIPTION
**Pull request by givemedeath**

### CODE REVIEW

File: SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs [hunk 1]
```diff
@@ -180,5 +87,37 @@ private static void ListEmotesCommand(ChatCommandBuilder builder)
...
+                        {
+                            return "You may only use the stuck command every five minutes. Please wait and try again.";
+                        }                        
+                    }
+
+                    return string.Empty;
+                })
+                .Action((user, target, location, args) =>
+                {
+                    var nextStuckAllowed = DateTime.UtcNow.AddMinutes(5);
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs:113]: Let's increase this to 30 minutes if you don't mind.

File: SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs [hunk 2]
```diff
@@ -180,5 +87,37 @@ private static void ListEmotesCommand(ChatCommandBuilder builder)
...
+                    var waypoint = GetNearestObject(Core.NWScript.Enum.ObjectType.Waypoint,user);
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs:114]: Can you add a using statement to the top so you only need to specify ObjectType.Waypoint here? The namespacing gets hard to read.

**Review by zunath — CHANGES_REQUESTED** [review]

File: SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs (hunk 1 above)
Comment (givemedeath) [reply SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs]: Done

File: SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs (hunk 2 above)
Comment (givemedeath) [reply SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs]: Done

**Comment by givemedeath** [comment]
Added initial implementation of subdual mode.

File: SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs [hunk 3]
```diff
@@ -45,105 +44,14 @@ private static void BugCommand(ChatCommandBuilder builder)
                         {
                             return "You may only submit one bug report per minute. Please wait and try again.";
                         }
+                        SetLocalString(user, "BUG_REPORT_LAST_SUBMISSION", dateTime.ToString());
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs:47]: It might make more sense to do this check when they submit on the UI. As it is now, if they run this command the timer starts as soon as they enter the command. If the user decides to cancel they need to wait a minute before they can open it again. I'd probably get rid of this validation section and do it on the UI's view model for the OnClick event. What do you think?

File: SWLOR.Game.Server/Feature/GuiDefinition/ViewModel/SettingsViewModel.cs [hunk 4]
```diff
@@ -82,6 +92,9 @@ private void UpdateHolonetSetting()
         {
             SetLocalBool(Player, "DISPLAY_HOLONET", DisplayHolonetChannel);
         }
-
+        private void UpdateSubdualMode()
+        {
+            SetLocalBool(Player, "SUBDUAL_MODE", SubdualMode);
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/GuiDefinition/ViewModel/SettingsViewModel.cs:97]: I wouldn't bother with the local variable. The only reason we do it with the Holonet is because of how frequent it is and how many players are involved (would require an indeterminate number of DB calls). I would get rid of this and change your check in Death.cs to check the player DB flag instead.

File: SWLOR.Game.Server/Service/Death.cs [hunk 5]
```diff
@@ -28,10 +28,27 @@ public static void OnPlayerDeath()
                 factionMember = GetNextFactionMember(hostile, false);
             }
 
-            const string RespawnMessage = "You have died. You can wait for another player to revive you or respawn to go to your home point.";
-            PopUpDeathGUIPanel(player, true, true, 0, RespawnMessage);
+            if (GetLocalBool(hostile, "SUBDUAL_MODE"))
```
Comment (zunath) [inline SWLOR.Game.Server/Service/Death.cs:31]: As mentioned in another comment, I'd recommend doing a check here for the DB entity's SubdualMode instead of a local variable.

**Comment by zunath** [comment]
This one's ok but if you don't mind - can you do separate branches and Pull Requests per feature? Getting hard to follow because there are three different tickets happening here. Really appreciate the help by the way!

File: SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs (hunk 3 above)
Comment (givemedeath) [reply SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs:47]: Good call, it was actually being set in OnClick event already anyway.

**Comment by givemedeath** [comment]
> This one's ok but if you don't mind - can you do separate branches and Pull Requests per feature? Getting hard to follow because there are three different tickets happening here. Really appreciate the help by the way!

Sure, I will try that.

File: SWLOR.Game.Server/Service/Death.cs (hunk 5 above)
Comment (givemedeath) [reply SWLOR.Game.Server/Service/Death.cs]: Done

File: SWLOR.Game.Server/Feature/GuiDefinition/ViewModel/SettingsViewModel.cs (hunk 4 above)
Comment (givemedeath) [reply SWLOR.Game.Server/Feature/GuiDefinition/ViewModel/SettingsViewModel.cs]: Done

File: SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs (hunk 3 above)
Comment (givemedeath) [reply SWLOR.Game.Server/Feature/ChatCommandDefinition/SystemChatCommand.cs]: Done

**Review by zunath — APPROVED** [review]
